components/infobox/wikis/halo/infobox_map_custom.lua

```diff
@@ -0,0 +1,90 @@
+---
+-- @Liquipedia
+-- wiki=halo
+-- page=Module:Infobox/Map
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local Map = require('Module:Infobox/Map')
+local Widgets = require('Module:Infobox/Widget/All')
+local Injector = require('Module:Infobox/Widget/Injector')
+local Cell = require('Module:Infobox/Widget/Cell')
+local String = require('Module:StringUtils')
+local MapModes = require('Module:MapModes')
+
+local CustomMap = Class.new()
+
+local CustomInjector = Class.new(Injector)
+
+local _args
+local _game
+
+local _GAME = mw.loadData('Module:GameVersion')
+
+function CustomMap.run(frame)
+	local customMap = Map(frame)
+	customMap.createWidgetInjector = CustomMap.createWidgetInjector
+	customMap.getCategories = CustomMap.getCategories
+	_args = customMap.args
+	return customMap:createInfobox(frame)
+end
+
+function CustomMap:createWidgetInjector()
+	return CustomInjector()
+end
+
+function CustomInjector:addCustomCells(widgets)
+	--[[table.insert(widgets, Cell{
+		name = 'Game',
+		content = {_args.game},
+		options = { makeLink = true }
+	})]]
```

Comment:
Why is this commented out? Also I feel like this could be added to the base

Reply:
Commented out to test the version in ``CustomInjector:parse(id, widgets)``. Will move this and the Release date to a PR on the base

Reply:
i disagree with moving that to /Base
game is used as an alias param (due to historical reasons) for the css class adjusting of the infobox on multi game wikis

![0A17C32D-4F60-4683-A886-FB87669D7DA1](https://user-images.githubusercontent.com/75081997/147425820-e8e642b2-1605-4934-b419-3fa15a717396.jpeg)
![547397CE-DD01-4F5E-8DE4-E36313B7E20A](https://user-images.githubusercontent.com/75081997/147425824-0a959128-96e8-4fa9-8596-b9478c51445e.jpeg)
![58408443-0A05-4DE4-B853-C5093853CAD4](https://user-images.githubusercontent.com/75081997/147425827-3524cc53-3df3-4714-9802-10b529f71d22.jpeg)

Reply:
So my next question will be what should the handling be for different games. Lets say we have this infobox on the counterstrike wiki, but the specific map is for CounterStrike Source, what do we do at that point? Do we keep a game field but rename it so it doesn't conflict or do we find a new way to display that information?

Reply:
1) If a wiki has a display it usually has to be converted because it is entered as an abbreviation. For that a conversion table is needed, which is wiki specific.
2) Most wikis do not have a display but still set it (often with a default value) for the css. I checked 24 wikis and found 18 without a display while setting the css with it and several of them using it for categories (again checking against a wiki specific conversion table). So i think it is safe to assume that the default should be to not display it. And that if it is to be displayed it has to be via the wiki specific part.

Reply:
Dota, Counterstrike, Siege, COD, Halo, AOE, Overwatch (if OW2 ever gets released), Fighting Games, and FIFA, are all wikis (that i can think of) that could benifit from having a game type (or version) in the infobox. 9 Wikis total compared to the 42 installed

Reply:
well there are 23 wikis that currently have the template (https://liquipedia.net/commons/Tools)

Reply:
Let's keep game/gameversion in custom for now at least then.

Reply:
as mentioned on PR: #886, Game/GameVersion has been removed from that PR and will be implemented here instead

Reply:
Marking this conversation as resolved. Current implementation of this code approved already and code on the base version has already been removed